src/Security/PermissionSet/Repository/PermissionSet.php

```diff
@@ -126,45 +124,51 @@ public function selectOneById(int $id, int $uid): Entity\PermissionSet
 	 * @param int $uid User id whom the items belong, used for ownership check.
 	 *
 	 * @return Collection\PermissionSets
+	 *
+	 * @throws PermissionSetPersistenceException
 	 */
 	public function selectByContactId(int $cid, int $uid): Collection\PermissionSets
 	{
-		$cdata = Contact::getPublicAndUserContactID($cid, $uid);
-		if (!empty($cdata)) {
-			$public_contact_str = $this->aclFormatter->toString($cdata['public']);
-			$user_contact_str   = $this->aclFormatter->toString($cdata['user']);
-			$cid                = $cdata['user'];
-		} else {
-			$public_contact_str = $this->aclFormatter->toString($cid);
-			$user_contact_str   = '';
-		}
+		try {
+			$cdata = Contact::getPublicAndUserContactID($cid, $uid);
+			if (!empty($cdata)) {
+				$public_contact_str = $this->aclFormatter->toString($cdata['public']);
+				$user_contact_str   = $this->aclFormatter->toString($cdata['user']);
+				$cid                = $cdata['user'];
+			} else {
+				$public_contact_str = $this->aclFormatter->toString($cid);
+				$user_contact_str   = '<<>>';
```

Comment:
I don't understand why we need this value? We shouldn't be able to generate it with `aclFormatter->toString` at all in the first place.

Reply:
The `REGEXP` call doesn't work with empty values with `mysql 5.7.36`, so we need explicit for this bug a fix. I don't want to alter the `aclFormatter->toString()` logic, because it seems overkill (And currently, an empty `$item` would lead to an empty output, so it wouldn't fix the behavior)

Reply:
The solution is to fix both the formatter method and stop using `REGEXP` since we are just looking for a substring, so something like `LOCATE` would be more accurate while maybe not choking on empty patterns?

Reply:
We don't search for a specific substring, because parts of the ACL could be anywhere in the corresponding ACL-column, so we would have to split and search every ACL als substring of the ACL column, I think that's why we use the `REGEXP` :-)

Reply:
ah, nevermind, for the `allow_cid` and `deny_cid`, there's only one value possible. I was just focusing on the `gid`ACLs

Reply:
Replaced the `REGEXP` with `LOCATE` for `allow_cid` and `deny_cid`.
I don't think that the `AclFormatter` needs any fixings, it's just because of the incompatibility of `mysql 5.7.36`.

Reply:
I'm still wondering how the `<<>>` values were generated.